### PR TITLE
feat: Allow cancellation & reschedule on past bookings

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Allow bookings in the past to be canceled or rescheduled to account for accurate booking "completion" metrics. 

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/modules/bookings/views/bookings-single-view.tsx [hunk 1]
```diff
@@ -466,7 +464,7 @@ export default function Success(props: PageProps) {
                             ? t("no_longer_attending")
                             : t("event_cancelled")
                           : isBookingInPast
-                          ? t("event_expired")
+                          ? t("event_is_in_the_past")
```
Comment (Udit-takkar) [inline apps/web/modules/bookings/views/bookings-single-view.tsx:467]:
```
 {needsConfirmation && isReschedulable
                          ? props.recurringBookings
                            ? t("booking_submitted_recurring")
                            : t("booking_submitted")
                          : isCancelled
                          ? seatReferenceUid
                            ? t("no_longer_attending")
                            : t("event_cancelled")
                          : isBookingInPast
                          ? t("event_expired")
                          : props.recurringBookings
                          ? t("meeting_is_scheduled_recurring")
                          : t("meeting_is_scheduled")
```         
                
can we simplify this code?

Comment (alishaz-polymath) [reply apps/web/modules/bookings/views/bookings-single-view.tsx:467]: Yeah seems like a good opportunity to make it more readable. Will take care of this 🙏

File: packages/features/calendars/DatePicker.tsx [hunk 2]
```diff
@@ -291,8 +291,6 @@ const Days = ({
         </div>
       ))}
 
-      {isBookingInPast && <ReschedulingNotPossibleOverlay />}
```
Comment (CarinaWolli) [inline packages/features/calendars/DatePicker.tsx:294]: we can remove the component `ReschedulingNotPossibleOverlay` in line 132

File: apps/web/components/booking/BookingListItem.tsx [hunk 3]
```diff
@@ -526,6 +526,7 @@ function BookingListItem(booking: BookingItemProps) {
             </>
           ) : null}
           {isBookingInPast && isPending && !isConfirmed ? <TableActions actions={bookedActions} /> : null}
+          {isBookingInPast && isConfirmed ? <TableActions actions={bookedActions} /> : null}
```
Comment (CarinaWolli) [inline apps/web/components/booking/BookingListItem.tsx:529]: Can you add some space between 'Edit' and 'Check for recordings'
Screenshot 2024-08-08 at 9 30 58 AM

Also, mobile view ist broken
Screenshot 2024-08-08 at 9 30 20 AM

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**